core/src/main/java/hudson/model/Actionable.java

```diff
@@ -91,18 +95,51 @@
      */
     @Exported(name="actions")
     public final List<? extends Action> getAllActions() {
-        List<Action> _actions = new ArrayList<Action>(getActions());
-        for (TransientActionFactory<?> taf : ExtensionList.lookup(TransientActionFactory.class)) {
-            if (taf.type().isInstance(this)) {
-                try {
-                    _actions.addAll(createFor(taf));
-                } catch (Exception e) {
-                    LOGGER.log(Level.SEVERE, "Could not load actions from " + taf + " for " + this, e);
+        List<Action> _actions = getActions();
+        boolean adding = false;
+        synchronized (Actionable.class) {
```

Comment:
🐛 ❗️ 💥   We're synchronizing every single getAllActions call on this single class.  Instant lock contention all over the place.

Synchronize on this.getClass() I think.  If we can avoid synchronization at all we should, though.